freqtrade/misc.py

```diff
@@ -262,7 +262,10 @@ def dataframe_to_json(dataframe: pandas.DataFrame) -> str:
     :param dataframe: A pandas DataFrame
     :returns: A JSON string of the pandas DataFrame
     """
-    return dataframe.to_json(orient='split')
+    # https://github.com/pandas-dev/pandas/issues/24889
+    # https://github.com/pandas-dev/pandas/issues/40443
+    # We need to convert to a dict to avoid mem leak
+    return dataframe.to_dict(orient='tight')
```

Comment:
i think we could/should use the following here:

``` python
def dataframe_to_json(dataframe: pandas.DataFrame) -> str:
    """
    Serialize a DataFrame for transmission over the wire using JSON
    :param dataframe: A pandas DataFrame
    :returns: A JSON string of the pandas DataFrame
    """
    # https://github.com/pandas-dev/pandas/issues/24889
    # https://github.com/pandas-dev/pandas/issues/40443
    # We need to convert to a dict to avoid mem leak
    def default(z):
        if isinstance(z, Timestamp):
            return z.timestamp() * 1e3
        raise TypeError

    return str(orjson.dumps(dataframe.to_dict(orient='split'), default=default), 'utf-8')
```

this will keep the consumers compatible (no change in the message formatting).